src/coreclr/pal/src/map/virtual.cpp

```diff
@@ -2140,7 +2141,17 @@ void ExecutableMemoryAllocator::TryReserveInitialMemory()
     int32_t preferredStartAddressIncrement;
     UINT_PTR preferredStartAddress;
     UINT_PTR coreclrLoadAddress;
-    const int32_t MemoryProbingIncrement = 128 * 1024 * 1024;
+
+#if TARGET_XARCH
+    const int32_t AddressProbingIncrement = 128 * 1024 * 1024;
+    const int32_t AllocSizeProbingDecrement = 128 * 1024 * 1024;
+#else
+    // Smaller steps on ARM becuase we try hard finding a spare memory in a 128Mb
+    // distance from coreclr so e.g. all calls from corelib to coreclr could use relocs
+    const int32_t AddressProbingIncrement = 8 * 1024 * 1024;
+    const int32_t AllocSizeProbingDecrement = 64 * 1024 * 1024;
```

Comment:
Why does the increment and decrement differ for arm?

Reply:
Unified (and increased the initial size to 1Gb) - works pretty good on my Apple M1 and even better on linux-ampere (more hits of successful reservations)

Reply:
in the worst case it does 8 probes (decreasing 1Gb by 128Mb each iteration and increasing startAddress by 8mb) and bails out